### ISSUE TITLE
Section number doesn't appear correctly

### ISSUE THREAD

**Issue by emitanaka** [rjournal/rjtools#53]
The Rmd file below displays the Section labels as `??` in the output HTML file (shown below) although the hyperlink is correct. 

![Screen Shot 2022-12-12 at 11 53 56 am](https://user-images.githubusercontent.com/7620319/206939952-dfb815ba-b7d4-4a92-978b-0ac8788fe33b.png)

For the PDF output (shown below), it shows the section number in the cross reference but there is no section number.

<img width="415" alt="Screen Shot 2022-12-12 at 11 51 17 am" src="https://user-images.githubusercontent.com/7620319/206939785-ba2c8eca-c0e9-4498-b140-c1c4c2784982.png">

````
---
title: "Test article"
date: "2022-11-09"
abstract: >
  An abstract of less than 150 words.
draft: true
author:  
  # see ?rjournal_article for more information
  - name: Quietest Quokka
type: package
output: 
  rjtools::rjournal_web_article:
    self_contained: yes
    toc: no
bibliography: RJreferences.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)
```

# Introduction

Refer to Section \@ref(bg)

# Background {#bg}

Test

```{r}
sessioninfo::session_info()
```
````

<details><summary>Session info</summary>

```
─ Session info ─────────────────────────────────────────────────────
 setting  value
 version  R version 4.1.2 (2021-11-01)
 os       macOS Big Sur 10.16
 system   x86_64, darwin17.0
 ui       X11
 language (EN)
 collate  en_AU.UTF-8
 ctype    en_AU.UTF-8
 tz       Australia/Melbourne
 date     2022-12-12
 pandoc   2.19.2 @ /Applications/RStudio.app/Contents/MacOS/quarto/bin/tools/ (via rmarkdown)

─ Packages ─────────────────────────────────────────────────────────
 package     * version    date (UTC) lib source
 bslib         0.4.1      2022-11-02 [1] CRAN (R 4.1.2)
 cachem        1.0.6      2021-08-19 [1] CRAN (R 4.1.0)
 cli           3.4.1      2022-09-23 [1] CRAN (R 4.1.2)
 cranlogs      2.1.1      2019-04-29 [1] CRAN (R 4.1.0)
 digest        0.6.30     2022-10-18 [1] CRAN (R 4.1.2)
 distill       1.5        2022-09-07 [1] CRAN (R 4.1.2)
 downlit       0.4.2      2022-07-05 [1] CRAN (R 4.1.2)
 evaluate      0.18       2022-11-07 [1] CRAN (R 4.1.2)
 fansi         1.0.3      2022-03-24 [1] CRAN (R 4.1.2)
 fastmap       1.1.0      2021-01-25 [1] CRAN (R 4.1.0)
 fs            1.5.2      2021-12-08 [1] CRAN (R 4.1.0)
 glue          1.6.2      2022-02-24 [1] CRAN (R 4.1.2)
 htmltools     0.5.4      2022-12-07 [1] CRAN (R 4.1.2)
 httr          1.4.4      2022-08-17 [1] CRAN (R 4.1.2)
 hunspell      3.0.2      2022-09-04 [1] CRAN (R 4.1.2)
 jquerylib     0.1.4      2021-04-26 [1] CRAN (R 4.1.0)
 jsonlite      1.8.4      2022-12-06 [1] CRAN (R 4.1.2)
 knitr         1.41       2022-11-18 [1] CRAN (R 4.1.2)
 lifecycle     1.0.3      2022-10-07 [1] CRAN (R 4.1.2)
 magrittr      2.0.3      2022-03-30 [1] CRAN (R 4.1.2)
 memoise       2.0.1      2021-11-26 [1] CRAN (R 4.1.0)
 purrr         0.3.5      2022-10-06 [1] CRAN (R 4.1.2)
 R6            2.5.1      2021-08-19 [1] CRAN (R 4.1.0)
 Rcpp          1.0.9      2022-07-08 [1] CRAN (R 4.1.2)
 rjtools       1.0.9.9000 2022-12-12 [1] Github (rjournal/rjtools@74f6cb0)
 rlang         1.0.6      2022-09-24 [1] CRAN (R 4.1.2)
 rmarkdown     2.18       2022-11-09 [1] CRAN (R 4.1.2)
 rstudioapi    0.14       2022-08-22 [1] CRAN (R 4.1.2)
 sass          0.4.4      2022-11-24 [1] CRAN (R 4.1.2)
 sessioninfo   1.2.2      2021-12-06 [1] CRAN (R 4.1.0)
 stringi       1.7.8      2022-07-11 [1] CRAN (R 4.1.2)
 stringr       1.5.0      2022-12-02 [1] CRAN (R 4.1.2)
 vctrs         0.5.1      2022-11-16 [1] CRAN (R 4.1.2)
 withr         2.5.0      2022-03-03 [1] CRAN (R 4.1.2)
 xfun          0.35       2022-11-16 [1] CRAN (R 4.1.2)
 yaml          2.3.6      2022-10-18 [1] CRAN (R 4.1.2)
 yesno         0.1.2      2020-07-10 [1] CRAN (R 4.1.0)

 [1] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
```
</details>

**Comment by emitanaka** [comment]
Note adding `number_section: yes` didn't show section numbers.

```
output: 
  rjtools::rjournal_web_article:
    self_contained: yes
    toc: no
    number_section: yes
```

**Comment by emitanaka** [comment]
Related issue #39

**Comment by mitchelloharawild** [comment]
The HTML referencing is an upstream issue with the {distill} package: https://github.com/rstudio/distill/issues/287
The PDF section numbering is an issue that could/should be fixed in {rjtools}.

**Comment by mitchelloharawild** [comment]
Also note that historically the R Journal has not used numbered sections, and it has been standard to refer to sections by name rather than number. This was more of a technical limitation from the process used to create a single PDF for published issues, which no longer applies to the new implementation in {rjtools}, however as you experience there are still some issues with referencing these numbered sections.

For now I suggest referring to sections by name.

**Comment by zeileis** [comment]
I also struggled with this, especially for getting hyperlinks to different sections in a longer paper. The workaround I used now is to provide the section number explicitly. Then the following works and is convenient for the readers:

    Refer to Section&nbsp;[2](#bg)

This works correctly in both PDF and HTML. (Additionally, the non-breaking space avoides line breaks in the middle of the section reference.) However, it is somewhat inconvenient to the authors because they need to remember to update the section number `2` if the section is moved back or forth.

P.S.: The new section numbering style hasn't propagated everywhere which might confusion (it did so for me and my co-author). I'll report this as a separate issue, though.
